easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -722,6 +722,26 @@ def get(self, key, default=None):
         else:
             return default
 
+    def __eq__(self, ec):
+        """Is this EasyConfig instance equivalent to the provided one?"""
+        return self.asdict() == ec.asdict()
+
+    def __hash__(self):
+        """Return hash value for a hashable representation of this EasyConfig instance."""
+        def make_hashable(val):
+            """Make a hashable value of the given value."""
+            if isinstance(val, list):
+                val = tuple([make_hashable(x) for x in val])
+            elif isinstance(val, dict):
+                val = tuple([(key, make_hashable(val)) for (key, val) in sorted(val.items())])
+            return val
+
+        tup = ()
+        for (key, val) in sorted(self.asdict().items()):
+            tup += (key, make_hashable(val))
```

Comment:
this is constructing quite a bit of tuples internally (i.e., it's slow), and there's probably no need to...

we should just use a list here?

```python
hashable = []
for (key, val) in sorted(self.asdict().items()):
    hashable.extend([key, make_hashable(val)])

return hash(hashable)
```

if the tests I added in https://github.com/ocaisa/easybuild-framework/pull/17 still pass with that in place, it's OK